### PR TITLE
Add compression feature

### PR DESCRIPTION
**Pull request by q-litzler**
Fixes #82.

### CODE REVIEW

File: src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt [hunk 1]
```diff
@@ -44,11 +47,25 @@ internal fun HttpClientConfig<*>.configure(configuration: Configuration) {
...
+    }
+}
+
+internal fun HttpMethod.canCompress(): Boolean {
+    return this == HttpMethod.Post || this == HttpMethod.Put
+}
+
+internal fun HttpRequestBuilder.compressionHeader(compression: Compression) {
+    when (compression) {
+        Compression.Gzip -> header(HttpHeaders.ContentEncoding, "gzip")
```
Comment (Ant-hem) [inline src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: Question: is the HTTP client setting the `Content-Length` header or do you have to set explicitly? (it's the case in Java)

File: src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt [hunk 2]
```diff
@@ -44,11 +47,25 @@ internal fun HttpClientConfig<*>.configure(configuration: Configuration) {
...
+        else -> Unit
     }
 }
```
Comment (Ant-hem) [inline src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: nit: could be nice to open another PRs to have all files finishing with a new line.

Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: Do you think it would be useful to have a new line at the end of each file ?

Comment (Ant-hem) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: It's a [posix standard convention ](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)I have seen in many projects. It's up to you :)

File: src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt (hunk 1 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: No it doesn't do it automatically, should I pass this header all the time, or only when doing gzip ?

File: src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt (hunk 2 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: I might do it then :) Thanks

File: src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt (hunk 1 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: Answer is: always.

Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: My bad: Actually the http engine handles it.

Comment (Ant-hem) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: All good!

File: src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt [hunk 3]
```diff
@@ -44,11 +47,25 @@ internal fun HttpClientConfig<*>.configure(configuration: Configuration) {
         logger = Logger.SIMPLE
     }
     install(UserAgent) {
-        agent = "Algolia for Kotlin (${BuildConfig.version})"
+        agent =  "Algolia for Kotlin (${BuildConfig.version})"
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: Extra space
```suggestion
        agent = "Algolia for Kotlin (${BuildConfig.version})"
```

Comment (PLNech) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: Besides this nitpick, LGTM :ok_hand:

Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt]: Fixed in subsequent PR 👌
